### PR TITLE
feat: merge present proof v2/v3 client APIs

### PR DESCRIPTION
**Pull request by Moopli**
Signed-off-by: Filip Burlacu <filip.burlacu@securekey.com>

### CODE REVIEW

**Review by fqutishat — APPROVED** [review]

File: pkg/wallet/wallet.go [hunk 1]
```diff
@@ -697,15 +708,28 @@ func (c *Wallet) Connect(authToken string, invitation *outofband.Invitation, opt
...
+func (c *Wallet) ProposePresentation(authToken string, invitation *GenericInvitation, options ...InitiateInteractionOption) (*service.DIDCommMsgMap, error) { //nolint: lll
 	opts := &initiateInteractionOpts{}
 	for _, opt := range options {
 		opt(opts)
 	}
 
-	connID, err := c.Connect(authToken, invitation, opts.connectOpts...)
-	if err != nil {
-		return nil, fmt.Errorf("failed to perform did connection : %w", err)
+	var connID string
```
Comment (sudeshrshetty) [inline pkg/wallet/wallet.go:717]: Looks good, 
but as discussed earlier, may be in some follow up PR, try to put these V1 vs V2 logic inside a wrapper where wallet just initializes right wrapper by passing `GenericInvitation`.

File: pkg/controller/rest/presentproof/operation.go [hunk 2]
```diff
@@ -53,7 +52,7 @@ type Operation struct {
 }
 
 // New returns new present proof rest client protocol instance.
-func New(ctx client.Provider, notifier command.Notifier) (*Operation, error) {
+func New(ctx presentproof.Provider, notifier command.Notifier) (*Operation, error) {
```
[the remark this answers is not in the document]
Comment (Moopli) [reply pkg/controller/rest/presentproof/operation.go:55]: keeping rest endpoints for now, for compatibility, but the V2 and V3 versions of an endpoint go through the same merged controller API.

[the remark this answers is not in the document]
Comment (sudeshrshetty) [reply pkg/controller/rest/presentproof/operation.go:55]: again, REST clients will have to decide which one to call. Same problem that SDK/JS users were facing

[the remark this answers is not in the document]
Comment (Moopli) [reply pkg/controller/rest/presentproof/operation.go:55]: With the handlers merged, REST clients can use the v2 endpoints for v3, they don't need to worry about which version endpoint to use - we can switch over REST clients in a followup.

**Pull request merged**